src/main/java/ch/njol/skript/effects/EffSendBlockChange.java

```diff
@@ -18,100 +18,115 @@
+})
+@Examples({
+	"make player see block at player as dirt",
+	"make player see target block as campfire[facing=south]",
+	"make player see {_blocks::*} as obsidian without light updates"
+})
+@Since("2.2-dev37c, 2.5.1 (block data support), INSERT VERSION (multi-block method)")
+@RequiredPlugins("MC 1.19+ (multi-block method)")
 public class EffSendBlockChange extends Effect {
 
-	private static final boolean BLOCK_DATA_SUPPORT = Skript.classExists("org.bukkit.block.data.BlockData");
-	private static final boolean SUPPORTED =
-			Skript.methodExists(
-					Player.class,
-					"sendBlockChange",
-					Location.class,
-					Material.class,
-					byte.class
-			);
+	private static final boolean SUPPORT_MULTI_BLOCKS = Skript.methodExists(Player.class, "sendBlockChanges", CollectionUtils.array(Collection.class, boolean.class));
 
 	static {
 		Skript.registerEffect(EffSendBlockChange.class,
-				BLOCK_DATA_SUPPORT ? "make %players% see %blocks% as %itemtype/blockdata%" : "make %players% see %blocks% as %itemtype%"
-		);
+			SUPPORT_MULTI_BLOCKS ? "make %players% see %blocks% as %itemtype/blockdata% [nolightupdate:without light update[s]]" : "make %players% see %blocks% as %itemtype/blockdata%");
 	}
 
-	@SuppressWarnings("null")
 	private Expression<Player> players;
-
-	@SuppressWarnings("null")
 	private Expression<Block> blocks;
-
-	@SuppressWarnings("null")
 	private Expression<Object> as;
+	private boolean lightUpdates;
 	
 	@Override
 	@SuppressWarnings("unchecked")
-	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
-		if (!SUPPORTED) {
-			Skript.error("The send block change effect is not supported on this version. " +
-				"If Spigot has added a replacement method without magic values " +
-				"please open an issue at https://github.com/SkriptLang/Skript/issues " +
-				"and support will be added for it.");
-			return false;
-		}
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		players = (Expression<Player>) exprs[0];
 		blocks = (Expression<Block>) exprs[1];
 		as = (Expression<Object>) exprs[2];
+		lightUpdates = !parseResult.hasTag("nolightupdate");
 		return true;
 	}
 
 	@Override
-	protected void execute(Event e) {
-		Object object = this.as.getSingle(e);
+	protected void execute(Event event) {
+		Object object = as.getSingle(event);
+		Block[] blocks = this.blocks.getArray(event);
 		if (object instanceof ItemType) {
 			ItemType itemType = (ItemType) object;
-			for (Player player : players.getArray(e)) {
-				for (Block block : blocks.getArray(e)) {
-					itemType.sendBlockChange(player, block.getLocation());
+			if (SUPPORT_MULTI_BLOCKS && blocks.length > 1) {
+				for (Player player : players.getArray(event)) {
+					Location[] locations = Arrays.stream(blocks).map(Block::getLocation).toArray(Location[]::new);
```

Comment:
this should be done outside the player loop

Reply:
Can also use `this.blocks.stream(event)` instead of using `Arrays.stream`